### DISCUSSION TITLE
How to pass proxy through environment?

### DISCUSSION THREAD

**Discussion by miklosn** [nolar/kopf#822] [Q&A]
I'm trying to use proxies during local development, but it seems kopf is not using the popular `https_proxy`, `HTTPS_PROXY` and similar env vars. Any idea? Thanks!

```Request attempt #1/9 failed; will retry: GET https://x.x.x.x/apis -> ClientConnectorError(ConnectionKey(host='34.73.227.85', port=443, is_ssl=True, ssl=None, proxy=None, proxy_auth=None, proxy_headers_hash=3431603120553464316), TimeoutError(60, "Connect call failed ('x.x.x.x', 443)"))```

**Comment by nolar** [comment]
You are right, proxies are not supported at the moment. It seems, proxy support is explicitly disabled by default in `aiohttp`:

* https://docs.aiohttp.org/en/stable/client_advanced.html#proxy-support

> Contrary to the requests library, it won’t read environment variables by default. But you can do so by passing trust_env=True into aiohttp.ClientSession constructor for extracting proxy configuration from HTTP_PROXY or HTTPS_PROXY environment variables (both are case insensitive): …

Fixing this in Kopf seems to be easy ([here](https://github.com/nolar/kopf/blob/1.33.0/kopf/_cogs/clients/auth.py#L156-L163)). But I wonder why did they disable this feature by default. Is it secure enough in the isolated environments where operators work?

**Reply by nolar** [reply]
Okay, I think I have found the background story for the proxy env vars support in aiohttp: https://github.com/aio-libs/aiohttp/issues/529. It looks like an argument between established practices to always support this env vars out of the box vs. avoiding implicit usage of them in all cases as (it was) believed by aiohttp's author.

The only insecure usage comes from injected proxies. But since the operators' environments are fully controlled, it makes sense to parse the env vars. Especially since libraries like `pykube-ng` & `kubernetes` will use them too, out of the box.

@miklosn Would you like to make a PR to add this option to the linked part of the source code?
